ivadomed/utils.py

```diff
@@ -236,14 +241,42 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
         nib_im = nib.load(fname)
         data_im = nib_im.get_fdata()
         data_lst.append(data_im)
-        data_im_l, _ = label(data_im, structure=bin_struct)
+        data_im_l = []
+        for i in range(data_im.shape[-1]):
+            data_im_tmp, _ = label(data_im[..., i], structure=bin_struct)
+            data_im_l.append(data_im_tmp)
         data_l_lst.append(data_im_l)
         del nib_im
+    data_l_lst = np.array(data_l_lst)
```

Comment:
why not using direct np array manipulation instead of going through a list?

Reply:
It was previsouly a list so I moved with that before realising I needed to transpose it, which is more convenient with an array. I'll change that.

Reply:
It was a list to match data_lst (that stores the image and not the masks) [L240](https://github.com/ivadomed/ivadomed/blob/65037dffb4361a0686e02ec753caba4d095e39e3/ivadomed/utils.py#L240)  , but I guess we can use np.array for the whole process

Reply:
Manipulating array is a bit more complicated, because the initialisation needs to have the right shape. Since we have not loaded the image when we initialize the data_l_lst objects, the shape is unknown (we can do it, it just increase the number of IO in the code). 
see https://numpy.org/doc/stable/reference/generated/numpy.empty.html. I think going through a list is more effcient. I can change the name though.

Reply:
@Drulex do you have some thoughts about best practice here?

Reply:
Not exactly sure about this specific case since I am not familiar with the topic at hand, but it looks like the lists are used exclusively as containers for the actual arrays and labels which seems ok.